diff --git a/src/backend/libsisyphus.py b/src/backend/libsisyphus.py
index 9512621..b0c7f8e 100644
--- a/src/backend/libsisyphus.py
+++ b/src/backend/libsisyphus.py
@@ -87,7 +87,11 @@ def getPkgDeps(pkgList):
     for portageOutput in io.TextIOWrapper(portageExec.stdout, encoding="utf-8"):
         if "/" in portageOutput.rstrip():
             pkgDep = str(portageOutput.rstrip().split("]")[1].split("[")[0].strip("\ "))
-            if not "blocks" and not "uninstall" in pkgDep:
+            if "blocks" in pkgDep:
+                pass
+            elif "uninstall" in pkgDep:
+                pass
+            else:
                 pkgDeps.append(pkgDep)
     return pkgDeps
 
@@ -99,7 +103,11 @@ def getWorldDeps():
     for portageOutput in io.TextIOWrapper(portageExec.stdout, encoding="utf-8"):
         if "/" in portageOutput.rstrip():
             worldDep = str(portageOutput.rstrip().split("]")[1].split("[")[0].strip("\ "))
-            if not "blocks" and not "uninstall" in worldDep:
+            if "blocks" in worldDep:
+                pass
+            elif "uninstall" in worldDep:
+                pass
+            else:
                 worldDeps.append(worldDep)
     return worldDeps
 
